File: TeeNew/Builder2010/Gallery_Smooth.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Gallery_Smooth.h"
#include "TeeThemeEditor.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeGalleryPanel"
#pragma resource "*.dfm"
TGallerySmooth *GallerySmooth;
//---------------------------------------------------------------------------
__fastcall TGallerySmooth::TGallerySmooth(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TGallerySmooth::CheckBox2Click(TObject *Sender)
{
  ChartGalleryPanel1->View3D = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TGallerySmooth::CheckBox1Click(TObject *Sender)
{
  ChartGalleryPanel1->Smooth = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TGallerySmooth::FormCreate(TObject *Sender)
{
  ChartGalleryPanel1->CreateGalleryPage("Standard");
  ChartGalleryPanel1->NumCols = 3;
  ChartGalleryPanel1->NumRows = 2;
}
//---------------------------------------------------------------------------

void __fastcall TGallerySmooth::Button1Click(TObject *Sender)
{
  TChartThemeSelector *theme = new TChartThemeSelector(this);
  try
  {
    theme->Chart = ChartGalleryPanel1->SelectedChart;
    theme->CBScale->Checked = false;
    if (theme->ShowModal() == mrOk) ChartGalleryPanel1->UseTheme(theme->SelectedTheme());
  }
  __finally
  {
    delete theme;
  }
}
//---------------------------------------------------------------------------

File: TeeNew/Gallery_Smooth.pas
unit Gallery_Smooth;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QComCtrls, QStdCtrls, QExtCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ComCtrls, StdCtrls, ExtCtrls,
  {$ENDIF}
  Base, TeEngine, Series, TeeProcs, TeeGalleryPanel, 
  {$IFNDEF TEENOTHEMES}
  TeeThemeEditor,
  {$ENDIF}
  Chart;

type
  TGallerySmooth = class(TBaseForm)
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    ChartGalleryPanel1: TChartGalleryPanel;
    Button1: TButton;
    procedure CheckBox2Click(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$R *.dfm}

procedure TGallerySmooth.CheckBox2Click(Sender: TObject);
begin
  ChartGalleryPanel1.View3D:=CheckBox2.Checked
end;

procedure TGallerySmooth.CheckBox1Click(Sender: TObject);
begin
  ChartGalleryPanel1.Smooth:=CheckBox1.Checked;
end;

procedure TGallerySmooth.FormCreate(Sender: TObject);
begin
  inherited;

  ChartGalleryPanel1.CreateGalleryPage('Standard');
  ChartGalleryPanel1.NumCols:=3;
  ChartGalleryPanel1.NumRows:=2;
end;

procedure TGallerySmooth.Button1Click(Sender: TObject);
begin
  {$IFNDEF TEENOTHEMES}
  with TChartThemeSelector.Create(Self) do
  try
    Chart:=ChartGalleryPanel1.SelectedChart;
    CBScale.Checked:=False;

    if ShowModal=mrOk then
       ChartGalleryPanel1.UseTheme(SelectedTheme);
  finally
    Free;
  end;
  {$ENDIF}
end;

initialization
  RegisterClass(TGallerySmooth);
end.
